### ISSUE TITLE
Provide a convenience method for HttpServerRequest to collect all content.

### ISSUE THREAD

**Issue by kurzweil** [ReactiveX/RxNetty#430]
Sometimes a developer knows, upfront, that they need all the content of a request. A convenience method (getAllContent()?), in addition to the streaming version of the getContent() method, would be helpful.

**Comment by NiteshKant** [comment]
+1 this would be good to have both on `HttpServerRequest` and `HttpClientResponse`. In absence of this method, one can use the following code snippet to aggregate the content:

``` java
resp.getContent()
    .collect(Unpooled::compositeBuffer,
    (composite, buf) -> {
        int readable = buf.readableBytes();
        composite.addComponents(buf);
        composite.writerIndex(composite.writerIndex() + readable);
    })
```

**Comment by jamesgorman2** [comment]
I was looking at this (basic `Observable.Transformer`) and I can make sense of `<ByteBuf, ByteBuf>` collection, but `ContentSource<T>` is generic, so I can't make sense of collection of all possible emission types from it. 

Can this be treated like `HttpClientResponse#getContentAsServerSentEvents()` which is forcing the decoding to `ContentSource<ServerSentEvent>`? It looks to me like it can't since we never store the type anywhere.

I have a couple of transformers to deal with this (one is just your code above decorated a bit, the other is byte limiter to prevent non termination). If there isn't a sensible way to make this easy it might be worth while just including them or something like them in `common`.

https://github.com/jamesgorman2/RxNetty/commit/4e723f270b48a06d9149372848a662624532164c

**Comment by NiteshKant** [comment]
@jamesgorman2 you are correct that this only applies to `ByteBuf` content and I agree that since this is not generic, it is better to include it in a utility class and not the main API.

I like your approach in the code you linked here, although for bounded collection (max bytes), I would throw an error if the content exceeds the limit as opposed to doing a `takeUntil`. I think you can just have a single `Transformer` which can be used for both bounded and unbounded.

**Comment by jamesgorman2** [comment]
Good call. I'll tidy up the classes/tests. I'll also fix the naming/javadoc so it's nicer to use.

**Comment by NiteshKant** [comment]
Closed via #524 (Thanks @jamesgorman2 !!!)
